redisgraph/graph.py

```diff
@@ -177,6 +177,11 @@ def query(self, q, params=None, timeout=None, read_only=False):
         except redis.exceptions.ResponseError as e:
             if "wrong number of arguments" in str(e):
                 print("Note: RedisGraph Python requires server version 2.2.8 or above")
+            if "unknown command" in str(e):
+                # `GRAPH.RO_QUERY` is unavailable in older versions.
+                command[0] = "GRAPH.QUERY"
+                response = self.redis_con.execute_command(*command)
+                return QueryResult(self, response)
```

Comment:
Please check to see if `read_only is True` if so
 recall `return self.query(q, params, timeout, read_only=False)`

Reply:
That's fine but what if we call the `query` function in an environment that doesn't have redisgraph. That would end up in an infinite loop instead of returning the error.